### PR TITLE
2-Step Verification

### PR DESCRIPTION
**Pull request by wcjr**

### CODE REVIEW

File: dictionaries/mfa.definition.json [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+		"ko": "TODO"
+	},
+	"use_totp": {
+		"en": "Use my smartphone app instead",
+		"es": "TODO",
+		"fr": "TODO",
+		"ko": "TODO"
+	},
+	"use_backupcode": {
+		"en": "Use a backup code instead",
```
Comment (fillup) [inline dictionaries/mfa.definition.json]: Can/should we change this to "Use a printable code instead"? Or maybe "Use a printable backup code instead." to cover both concepts.

Comment (wcjr) [reply dictionaries/mfa.definition.json]: can't hurt to have both in there, thanks:  https://github.com/silinternational/simplesamlphp-module-material/pull/33/commits/97465395844113b622e302337375b69dfb819eac

File: README.md [hunk 2]
```diff
@@ -56,86 +56,153 @@ If configured, an alert will be shown to the user filled with the content of tha
...
...lexUzmp2Kl802KMiyzGuMhijIvYH3yRyq7iROWNip3KruIJlZOKncpJxRMqu4pvUtlV7FR2FScqJxU7lZOKNxZjXGQxxkUWY1zkw0squ4qdyhMV36Syq9ipPKFyUnGiclKxUzmpeKJip/JGxU7l/2kxxkUWY1xkMcZF7A++SGVXsVPZVexUnqg4UdlVnKjsKk5UTiqeUHmiYqdyUvFNKicVO5VdxTctxrjIYoyLLMa4yIeXVHYVO5VdxRMVJyq7im9SOak4UdlVvFHxRMU3qTyhcqKyq3hjMcZFFmNcZDHGRT58mcqu4kTlROWkYqeyqzhReaLim1SeUHlD5aTiiYoTlV3FLy3GuMhijIssxrjIh5cqdio7lV3FrmKnsqt4Q2VXcVKxUzlR2VW8UXGisqvYqTxRcaKyqzhROVE5qXhjMcZFFmNcZDHGRT78Yyp2KruKk4qdyknFTmVXsVN5Q+WNiidUnlDZVZyo7Cp2KicV37QY4yKLMS6yGOMiH15S2VXsVJ5Q2VWcqJxUfFPFTuWk4gmVXcVO5aRip/JExU7lCZUnVHYVbyzGuMhijIssxriI/cEPqfxSxRMqu4qdyhsVb6icVJyo/MsqfmkxxkUWY1xkMcZFPvxYxS+p7CqeUDmpeEJlV3Gisqs4UTmp2Kk8UfGEyr9kMcZFFmNcZDHGRT68pPI3VZyonFScqPw/qewqdhU7lZOKE5UTlV3FScUTKruKNxZjXGQxxkUWY1zkw5dVfJPKGxVvqJxU7CpOVHYVJyq7il3FicobFU+o7Cp2KruKb1qMcZHFGBdZjHGRDz+m8kTFExU7lZ3KScWu4ptUdhVvqDxRcaKyU/kvW4xxkcUYF1mMcZEP/3Equ4onVN6oeEPliYqdyq5ip7KreKNip3Kisqv4pcUYF1mMcZHFGBf5cDmVXcVJxRMqJxU7lZOKJypOKnYqb6icVOxUdiq7im9ajHGRxRgXWYxxkQ8/VvFLFTuVE5VdxU5lV/FNFTuVE5U3Kk4qTlROKnYqu4q/aTHGRRZjXGQxxkU+fJnK36TyRMUbKruKE5U3Kp5QeULlb1I5qXhjMcZFFmNcZDHGRewPxrjEYoyLLMa4yGKMiyzGuMhijIssxrjIYoyLLMa4yGKMiyzGuMhijIssxrjIYoyLLMa4yGKMi/wPXRyfsu/8YKUAAAAASUVORK5CYII=`
```
[the remark this answers is not in the document]
Comment (wcjr) [reply README.md]: right, not without our serverless db unfortunately...similar to our U2F constraint.

File: dictionaries/mfa.definition.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (wcjr) [reply dictionaries/mfa.definition.json]: thanks!

File: README.md [hunk 3]
```diff
@@ -56,86 +56,153 @@ If configured, an alert will be shown to the user filled with the content of tha
...
+
+#### Backup code
+1. Goto [http://ssp-hub.local](http://ssp-hub.local)
+2. Login as hub administrator: `username=`**admin** `password=`**abc123**
+3. Click **Authentication** tab
+4. Click **Test configured authentication sources**
+5. Click **hub-discovery**
+6. Click **idp4**
+7. Login as a "backup code" user: `username=`**has_backupcode** `password=`**a**
+8. Enter one fo the following codes to verify (`94923279, 82743523, 77802769, 01970541, 37771076, 39178450, 01813404, 00416843, 90068701, 35775442, 94923279, 82743523, 77802769, 01970541, 37771076, 39178450, 01813404, 00416843, 90068701, 35775442`)
```
Comment (Baggerone) [inline README.md]: "fo" -> "of"

Comment (wcjr) [reply README.md]: got it, thanks

File: themes/material/mfa/prompt-for-mfa-u2f.php [hunk 4]
```diff
@@ -0,0 +1,186 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <title><?= $this->t('{material:mfa:title}') ?></title>
```
[the remark this answers is not in the document]
Comment (wcjr) [reply themes/material/mfa/prompt-for-mfa-u2f.php]: nope, these will sometimes contain HTML.

**Pull request merged**